lib/comgr/test/CMakeLists.txt

```diff
@@ -73,3 +74,8 @@ add_comgr_test(isa_name_parsing_test)
 add_comgr_test(get_data_isa_name_test)
 add_comgr_test(include_subdirectory_test)
 add_comgr_test(options_test)
+# Test : Compile HIP only if HIP-Clang is installed.
+find_package( hip CONFIG PATHS /opt/rocm )
```

Comment:
I think this introduces a circular dependency between HIP and Comgr; is there any way to build all of these components from source anymore after this change?

Reply:
> I think this introduces a circular dependency between HIP and Comgr; is there any way to build all of these components from source anymore after this change?

The support of HIP compilation does not depend on HIP package. Only the test depends on it. I made changes for this check to be optional. If hip is not installed, comgr still builds successfully without error. Then HIP can be built with comgr.